### PR TITLE
GroupedList: Fix "Show All" not rendering all items in a section

### LINKED ISSUE #9561: GroupedList : 'Show All' does not work
### Environment Information 
 
- __Package version(s)__: 6.187.0

### Please provide a reproduction of the bug in a codepen:

'Show All' does not appear to work for GroupedList
https://codepen.io/anon/pen/gNmKWJ?editors=0010
 
#### Actual behavior:
 
Clicking 'Show All' does not expand the list.
 
#### Expected behavior:
 
It should expand the list

#### Priorities and help requested:
 
Are you willing to submit a PR to fix?

No
 
Requested priority:

Normal

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9561
- [x] Include a change request file using `$ npm run change`

#### Description of changes

`GroupedList` has the capability per section to render a "Show All" call to action that when pressed toggles a boolean on the `IGroup` instance. Currently when this happens, the "Show All" UI is dismissed, but the rendered rows remain the same. Instead, ALL of the rows should render.

Unfortunately it is not easy to track down the feature add and assumed regression via git history. However, this change fixes the issue.

**Note:** I will cherry-pick this fix into `6.0`.

Thanks @dliu120 for the initial bug report.

#### Focus areas to test

Compare the following screen-capture:

groupedlist_showall

with current behavior at: https://codepen.io/anon/pen/gNmKWJ?editors=0010

---

I will attempt to follow-up with unit test coverage. I tried initially, but there is something wrong with `GroupedList` in unit tests that requires debugging.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9610)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
FYI added test coverage in https://github.com/OfficeDev/office-ui-fabric-react/pull/9610/commits/458c437bd8daf70b09ae823f3483da17f9d36eb3 and https://github.com/OfficeDev/office-ui-fabric-react/pull/9610/commits/7a86696f90fd44d7fea2bad1d1b6ab0b71c71fc5

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.test.tsx [hunk 1]
```diff
@@ -56,4 +58,202 @@ describe('GroupedList', () => {
...
+
+    let listRows = wrapper.find(DetailsRow);
+    expect(listRows.length).toBe(1);
+
+    const groupShowAllElement = wrapper.find(GroupShowAll);
+
+    groupShowAllElement.simulate('click');
+
+    listRows = wrapper.find(DetailsRow);
+    expect(listRows.length).toBe(1);
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.test.tsx]: woops!

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.test.tsx]: Fixed in https://github.com/OfficeDev/office-ui-fabric-react/pull/9610/commits/5d6d4853ece1cc5526330eb30d73c9b9dabebc93!

**Comment by dliu120** [comment]
Thanks for the fix! Could we have this cherry-picked to v6? @KevinTCoughlin

**Comment by KevinTCoughlin** [comment]
> Thanks for the fix! Could we have this cherry-picked to v6? @KevinTCoughlin. No rush -- it'll be a while before we can upgrade to 7

@dliu120 it should have already been in https://github.com/OfficeDev/office-ui-fabric-react/pull/9625 which was released in `office-ui-fabric-react@v6.192.1`. Can you confirm?

**Comment by dliu120** [comment]
Oh even better! Thanks -- will try it out

**Comment by dliu120** [comment]
Works well TY- Found an issue with keyboard nav. Not sure if it's supposed to be handled by the consumer or the grouped list. Will do some investigation and file separately if needed.

**Pull request merged**